tensorflow/compiler/xla/service/slice_sinker.cc

```diff
@@ -0,0 +1,276 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/xla/service/slice_sinker.h"
+#include <algorithm>
+#include <utility>
+#include <vector>
+#include "absl/types/optional.h"
+#include "absl/container/flat_hash_set.h"
+#include "tensorflow/compiler/xla/shape_util.h"
+
+namespace xla {
+
+namespace {
+
+class SliceSinkerImpl {
+ public:
+  // Returns whether the instruction has been visited. The visited instruction
+  // has been grouped in a true user group with its peers and it should not be
+  // grouped in another true user group.
+  bool IsVisited(const HloInstruction* instruction) const;
+
+  // Collects true operands of inst. The inst's operands should all be slices,
```

Comment:
Can we replace the use of terms "true operands" and "true users" in the comments with "slice sources" and "peer operations" to match the code?